runs.py

```diff
@@ -194,6 +195,45 @@ def loop_cached_runs(*, is_prev: bool):
             loop_cached_runs(is_prev=False)
             return StreakInfo(streak_total, position_in_streak, is_ongoing)
         return StreakInfo(0, 0, False)
+    
+    @property
+    def start_time(self) -> datetime.datetime:
+        return self.timestamp - datetime.timedelta(seconds=int(self._data["playtime"]))
+
+    @property
+    def has_archive_link(self) -> bool:
+        return self.archive_link != ""
+
+    @property
+    def archive_link(self) -> str:
+        earliest_start = self.start_time
+        current_run = self.matched.get_run(is_prev=True, is_character_specific=False)
+        while current_run is not None and current_run.timestamp.date() == self.timestamp.date():
+            earliest_start = current_run.start_time
+            current_run = current_run.matched.get_run(is_prev=True, is_character_specific=False)
+        run_delta = self.start_time - earliest_start
+
+        date_str = f"({self.timestamp.date()})"
+        offset_re = re.compile(r"(\d?\d:\d\d:\d\d) Slay the Spire")
+
+        video_id = ""
+        video_offset = 0
+        with open("data/youtube.json", "r") as json_raw:
```

Comment:
Opening this each time feels pretty bad. Any preferences on where to put this?

Reply:
Oh yeah, this should be opened once on startup and saved in memory. Then, it should be checked (and updated) once a day (updating the JSON file in the process).

Reply:
Do we have a standard "run this once X time" built into the server? I hadn't noticed one but maybe.

Reply:
Nah, but this shouldn't be very hard to implement. Just using `Routine` from TwitchIO lets us run stuff at a given time every day.